mypy/server/deps.py

```diff
@@ -734,6 +749,43 @@ def visit_union_type(self, typ: UnionType) -> List[str]:
         return triggers
 
 
+def collect_protocol_attr_deps(names: SymbolTable,
+                               processed: Optional[Set[TypeInfo]] = None) -> Dict[str, Set[str]]:
+    """Recursively collect protocol attribute dependencies for classes in 'names'."""
+    # Currently there can be at most one target per trigger, but we might
+    # add more deps in future so we are using defaultdict.
+    if processed is None:
+        processed = set()
+    deps = defaultdict(set)  # type: DefaultDict[str, Set[str]]
+    for node in names.values():
+        if isinstance(node.node, TypeInfo) and node.node not in processed:
```

Comment:
Yes, this will improve speed a bit, will do.

Reply:
Note however that this means we should never skip processing `typing`. Because e.g. subtype cache of `typing.Iterable` also needs to be invalidated if something tested against it changed.